### PR TITLE
Automation for LabKey R&D UX - Field Editor Updates 2020, Summary View

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
1. Adding test coverage for toggle between Summary and detail mode
2. Links in the detail mode

#### Related Pull Requests
* https://github.com/LabKey/sampleManagement/pull/503
* https://github.com/LabKey/labkey-ui-components/pull/464
* https://github.com/LabKey/platform/pull/2066

#### Changes

1. src/org/labkey/test/components/domain/DomainFormPanel.java
2. src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java

### CODE REVIEW

File: src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java [hunk 1]
```diff
@@ -27,10 +26,14 @@
 import java.util.List;
 
 @Category(DailyA.class)
-@BaseWebDriverTest.ClassTimeout(minutes = 2)
+@BaseWebDriverTest.ClassTimeout(minutes = 3)
 public class FieldEditorRowSelectionActionTest extends BaseWebDriverTest
 {
     private final static String PROJECT_NAME = "Field Editor Row Selection Action Test";
+    private final static List<String> expectedHeaders = Arrays.asList("Name", "Range URI", "Required", "Lock Type", "Lookup Container", "Lookup Schema", "Lookup Query",
```
Comment (RosalineP) [inline src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java]: Should we include a test in a folder in which the `Ontology` module is not enabled, in order to ensure `"Source Ontology", "Concept Import Column", "Concept Label Column", "Principal Concept Code"` columns accurately do not get rendered, or would that be an overzealous test?

File: src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java [hunk 2]
```diff
@@ -83,6 +88,19 @@ public void testListActions() throws Exception
 
         checker().verifyTrue("Exported Fields are not same as UI", uiFields.equals(exportedFields));
 
+        log("Verifying the toggle between summary view and detail view list domain");
```
Comment (RosalineP) [inline src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java:90]: The section here, and the block at line 127, and at 200, have a lot of similarities and the differences they do have look like they could be adjusted via function arguments, if they were modularized into a function `testSummaryViewActions` or something.

Would it be good style to group these three chunks into one function, or in tests is it preferable to have the code be more explicit like this?

**Review by RosalineP — COMMENTED** [review]
Just a couple comments! Thanks for doing the tests on this 👍

Please leave merge task to me, as I will be going through the ui-components publishing process on these branches due to Product drift necessitating a Product update between feature merge and Automation Test merge.

File: src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java (hunk 1 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java]: Done.

File: src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java (hunk 2 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java:90]: I thought about it, it was mostly assert statements and not a lot of action having. Decided to keep them in the test.

Let me know if you feel strongly to move.

**Review by RosalineP — APPROVED** [review]
Looks good!

**Pull request merged**